frontend/js/src/common/UserSearch.tsx

```diff
@@ -14,97 +11,131 @@ export type UserSearchProps = {
   initialValue?: string;
 };
 
-export type UserSearchState = {
-  newUser: string;
-  userSearchResults: Array<SearchUser>;
-};
+export default function UserSearch(props: UserSearchProps) {
+  // Context
+  const { APIService } = React.useContext(GlobalAppContext);
 
-export default class UserSearch extends React.Component<
-  UserSearchProps,
-  UserSearchState
-> {
-  static contextType = GlobalAppContext;
-  declare context: React.ContextType<typeof GlobalAppContext>;
+  // Props
+  const { onSelectUser, placeholder, clearOnSelect, initialValue } = props;
 
-  // eslint-disable-next-line react/sort-comp
-  constructor(props: UserSearchProps) {
-    super(props);
-    this.state = {
-      newUser: props.initialValue ?? "",
-      userSearchResults: [],
-    };
-  }
+  // States
+  const [newUser, setNewUser] = React.useState(initialValue ?? "");
+  const [userSearchResults, setUserSearchResults] = React.useState<
+    Array<SearchUser>
+  >([]);
+  const [selectedIndex, setSelectedIndex] = React.useState(-1);
 
-  throttledSearchUsers = _throttle(async () => {
-    await this.searchUsers();
-  }, 300);
+  // Refs
+  const dropdownRef = React.useRef<HTMLSelectElement>(null);
 
-  searchUsers = async () => {
-    const { APIService } = this.context;
-    const { newUser } = this.state;
+  const searchUsers = async () => {
```

Comment:
We can memoize this function and pass it the user name to search as an argument:
```tsx
  const searchUsersFunction = React.useCallback(
    async (username: string) => {
      try {
        const response = await searchUsers(username);
        setUserSearchResults(response.users);
      } catch (error) {
        toast.error(<ToastMsg title="Error" message={error.message} />, {
          toastId: "error",
        });
      }
    },
    [searchUsers]
  );
```

And below also memoize the throttled function:
```jsx
const throttledSearchUsers = React.useMemo(
    () => _throttle(searchUsersFunction, 300),
    [searchUsersFunction]
  );
```

Note that I deconstructed `searchUsers` from APIService and as a consequence I renamed the function to `searchUsersFunction`. Naming can probably be improved